File: 206. 反转链表.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    # 三指针法 
    def reverseList(self, head):
        """
        :type head: Optional[ListNode]
        :rtype: Optional[ListNode]
        """
        curr = head
        pre = None
        while curr:
            n = curr.next
            curr.next = pre
            pre = curr
            curr = n
        return pre

File: 206. 反转链表.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func reverseList(head *ListNode) *ListNode {
	if head == nil {
		return nil
	}

	dummy := &ListNode{}
	curr := head

	for curr != nil {
		n := curr.Next
		curr.Next = dummy.Next
		dummy.Next = curr
		curr = n
	}

	return dummy.Next
}